serve/mlc_serve/engine/engine_common.py

```diff
@@ -302,18 +303,25 @@ def check_prompt_too_long(self, prompt_len: int, num_sequences: int = 1) -> bool
             or (kv_cache_size - prompt_len) < self.max_decode_steps * num_sequences
         )
 
-    def evict_request(self) -> int:
+    def evict_request(self, cancell_callback: Callable[[RequestId], None]) -> int:
+        # Must be called with the queue lock held
         num_eviction = 0
 
         while self.cache_manager.get_max_new_tokens() < 1:
             num_eviction += 1
             request_to_remove = min(
                 self.current_batch.values(), key=lambda s: s.num_total_tokens
             )
-            # TODO parallel sampling: Properly support evicting a multi-sequence request
-            assert (
-                self.current_batch[request_to_remove.request_id].num_sequences == 1
-            ), "Evicting a multi-sequence request is not supported."
+
+            # TODO(masahi): Properly support evicting a multi-sequence request
```

Comment:
I'll follow-up with this today. This indeed a temp solution, but even if after have a proper one, it's still good to make the best effort to avoid evicting a parallel sampling request.